File: bench/template/maps/fill_sum/fill_sum.lua
local function fill_sum(n)
  local m = {}
  for i = 0, n - 1 do
    m[i] = i
  end
  local s = 0
  for j = 0, n - 1 do
    s = s + m[j]
  end
  return s
end

local n = {{ .N }}
local repeats = 1000
local last = 0

local start = os.clock()
for _ = 1, repeats do
  last = fill_sum(n)
end
local duration_us = (os.clock() - start) * 1e6

io.write(string.format('{"duration_us": %d, "output": %s}\n', math.floor(duration_us), tostring(last)))

File: bench/template/maps/fill_sum/fill_sum.py
import json
import time

def fill_sum(n):
    m = {}
    for i in range(n):
        m[i] = i
    s = 0
    for j in range(n):
        s += m[j]
    return s

n = {{ .N }}
repeat = 1000
last = 0

start = time.perf_counter()
for _ in range(repeat):
    last = fill_sum(n)
duration = (time.perf_counter() - start) * 1e6

print(json.dumps({
    "duration_us": duration,
    "output": last,
}))
